### PR TITLE
Move files into folders

### PR DESCRIPTION
**Pull request by mburridge**
Move faq.md and architecture.md into their own folders to clean up root folder and create clearer structure for future development of docs.

### CODE REVIEW

File: docs/SUMMARY.md [hunk 1]
```diff
@@ -29,7 +29,7 @@
   * [9. Styles](learning-frontity/styles.md)
   * [10. Code Splitting](learning-frontity/code-splitting.md)
   * [11. Head](learning-frontity/head.md)
-* [🏗 Architecture](architecture.md)
+* [🏗 Architecture](architecture/architecture.md)
```
Comment (juanmaguitar) [inline docs/SUMMARY.md]: Don't you have to change this link to `architecture/README.md`?

File: docs/SUMMARY.md [hunk 2]
```diff
@@ -57,5 +57,5 @@
 * [💻 Key Concepts](resources/README.md)
   * [JavaScript](resources/javascript-basics.md)
   * [React](resources/react-basic.md)
-* [🤔 FAQ](faq.md)
+* [🤔 FAQ](faq/faq.md)
```
Comment (juanmaguitar) [inline docs/SUMMARY.md]: Don't you have to change this link to `faq/README.md`

File: docs/SUMMARY.md (hunk 1 above)
Comment (mburridge) [reply docs/SUMMARY.md]: Already done

File: docs/SUMMARY.md (hunk 2 above)
Comment (mburridge) [reply docs/SUMMARY.md]: Already done

**Pull request merged**